js/interface/map-panel.js

```diff
@@ -56,7 +134,29 @@ Fliplet.InteractiveMap.component('map-panel', {
         }
       })
 
-      window.filePickerProvider.then((result) => {
+      window.filePickerProvider.then(result => {
+        if (this.decidedToKeepMarkers) {
```

Comment:
```js
if (!this.decidedToKeepMarkers) {
  return;
}

// rest of your code
```

as per code guidelines

Reply:
can you rename `decidedToKeepMarkers` to `shouldKeepMarkers`?

Reply:
First comment - cannot do as suggested because part of the code after if statement should be executed anyway.
Second comment - fixed.